netcore/System.Private.CoreLib/src/System/GC.cs

```diff
@@ -236,10 +237,23 @@ internal static ulong GetSegmentSize ()
 			return 1024 * 1024 * 16;
 		}
 
+
+		[MethodImplAttribute (MethodImplOptions.InternalCall)]
+		static extern GCMemoryInfo _GetGCMemoryInfo (out long highMemoryLoadThresholdBytes,
```

Comment:
What is this `GCMemoryInfo` return type?
I bet this is the mysterious extra parameter -- a pointer to receive a non-tiny struct by value.
Use it if needed. Remove if it not.
Imho somewhat prefer not passing or returning structs by value, of any construct, to avoid ABI edges.

Reply:
It should probably return `void`.
And again I think the params should be `ref` and not `out`, but opinions vary.

Reply:
Eliminating the unneeded return.

Re ref/out, I don't think it matters a lot since this is a private method no one else is likely to call. It looks a little clearner to use out.

Reply:
“ this is a private method no one else is likely to call” and “it looks a little cleaner”
are not convincing counterpoints to me to “programmers repeatedly get this wrong in the runtime” and “let the C# compiler help you”.
I think out params are mostly a relic of micro-optimization.